via/services/youtube_api/models.py

```diff
@@ -0,0 +1,118 @@
+from dataclasses import dataclass
+from typing import List, Optional
+
+
+@dataclass
+class VideoDetails:
+    id: str
+    title: str
+    short_description: str
+    author: str
+    thumbnails: List[dict]
+
+    @classmethod
+    def from_json(cls, data):
+        return VideoDetails(
+            id=data["videoId"],
+            title=data["title"],
+            short_description=data["shortDescription"],
+            author=data["author"],
+            thumbnails=data["thumbnail"]["thumbnails"],
+        )
+
+
+@dataclass
+class CaptionTrack:
+    id: str
+    name: str
+    language_code: str
+    kind: str
+    is_translatable: bool
+    url: str = None
+
+    @classmethod
+    def from_json(cls, data):
+        return CaptionTrack(
+            id=data["vssId"],
+            language_code=data["languageCode"],
+            name=data["name"]["simpleText"],
+            kind=data.get("kind", None),
+            is_translatable=data.get("isTranslatable", False),
+            url=data["baseUrl"],
+        )
+
+    @property
+    def is_auto_generated(self):
+        return self.kind == "asr"
+
+
+@dataclass
+class Captions:
+    tracks: List[CaptionTrack]
+    translation_languages: List[dict]
+
+    @classmethod
+    def from_json(cls, data):
+        return Captions(
+            tracks=[
+                CaptionTrack.from_json(track) for track in data.get("captionTracks", [])
+            ],
+            translation_languages=[
+                {"code": language["languageCode"], "name": language["languageName"]}
+                for language in data.get("translationLanguages", [])
+            ],
```

Comment:
We're not using it, but if:

 * A transcript `is_translatable`
 * And you have languages here
 * Then you can request a transcript be translated by appending `&tlang={language_code}` on the end